### PR TITLE
dont calculate the relative path for every node from scratch

### PR DESCRIPTION
**Pull request by icewind1991**
instead use the relative path we already have for the folder

Minor performance improvement

[comparison](https://blackfire.io/profiles/compare/383746f1-506a-49c3-817b-e216ee8cbef6/graph)

cc @rullzer

### CODE REVIEW

File: apps/dav/lib/connector/sabre/directory.php [hunk 1]
```diff
@@ -162,14 +162,16 @@ public function createDirectory($name) {
 	 *
 	 * @param string $name
 	 * @param \OCP\Files\FileInfo $info
+	 * @param string|null $relativePath
```
Comment (PVince81) [inline apps/dav/lib/connector/sabre/directory.php]: relative to what ? root ? user home ?

Comment (icewind1991) [inline apps/dav/lib/connector/sabre/directory.php]: added docs

File: apps/dav/lib/connector/sabre/directory.php [hunk 2]
```diff
@@ -162,14 +162,16 @@ public function createDirectory($name) {
...
 	 * @param \OCP\Files\FileInfo $info
+	 * @param string|null $relativePath path relative to the user home
 	 * @return \Sabre\DAV\INode
 	 * @throws InvalidPath
 	 * @throws \Sabre\DAV\Exception\NotFound
 	 * @throws \Sabre\DAV\Exception\ServiceUnavailable
 	 */
-	public function getChild($name, $info = null) {
-		$path = $this->path . '/' . $name;
+	public function getChild($name, $info = null, $relativePath = null) {
```
Comment (PVince81) [inline apps/dav/lib/connector/sabre/directory.php]: Does it mean that now I can call `getChild` and pass an arbitrary relative path, even if that path is outside the current directory ?

File: apps/dav/lib/connector/sabre/node.php [hunk 3]
```diff
@@ -64,10 +64,11 @@
 	 *
 	 * @param \OC\Files\View $view
 	 * @param \OCP\Files\FileInfo $info
+	 * @param string|null $relativePath path relative to the user home
 	 */
-	public function __construct($view, $info) {
+	public function __construct($view, $info, $relativePath = null) {
 		$this->fileView = $view;
-		$this->path = $this->fileView->getRelativePath($info->getPath());
+		$this->path = (!is_null($relativePath)) ? $relativePath : $this->fileView->getRelativePath($info->getPath());
```
Comment (DeepDiver1975) [inline apps/dav/lib/connector/sabre/node.php]: Wouldn't it make more sense to add getRelativePath() to FileInfo?

We could easily keep the information in the info object which is passed around anyhow

Comment (icewind1991) [inline apps/dav/lib/connector/sabre/node.php]: Relative path is depended on things like logged in user, I would prefer not to add that dependency to FileInfo

**Comment by DeepDiver1975** [comment]
rebased

File: apps/dav/lib/connector/sabre/directory.php (hunk 2 above)
Comment (PVince81) [inline apps/dav/lib/connector/sabre/directory.php]: @icewind1991 ping

**Comment by PVince81** [comment]
Impact unclear and unanswered comments, moving to 9.1

**Comment by PVince81** [comment]
CC @cmonteroluque

**Comment by PVince81** [comment]
How much was the improvement ?

**Comment by PVince81** [comment]
@icewind1991 will you continue this or should we close ? (no feedback)

**Comment by PVince81** [comment]
Closing due to lack of feedback and many unanswered questions.

**Pull request closed without merging**